src/main/java/com/zufar/onlinestore/common/ApiResponse.java

```diff
@@ -0,0 +1,10 @@
+package com.zufar.onlinestore.common;
+
+import lombok.Builder;
+import java.time.Instant;
+
+@Builder
+public record ApiResponse(String data,
```

Comment:
Name of "ApiResponse" tells me it is a generic class for every response in our application. 
I think It shoud be used everewhere even in PaymentEndpoint but PaymentEndpoint uses ResponseEntity.
I think this name is unclear and used only for error case and also "success" is always false. 
So may be good name for this class is ErrorApiResponse. But you mean that this class is generic it should be like this ApiResponse<T>(T data,...) and should be used instead of ResponseEntity

Reply:
Done